app/src/main/java/org/wikipedia/page/leadimages/LeadImagesHandler.java

```diff
@@ -134,40 +147,60 @@ private void updateCallToAction() {
-                                        captionSourcePageTitle.getDisplayText(), currentCaption, getLeadImageUrl());
-
-                                callToActionTargetSummary = new PageSummaryForEdit(captionTargetPageTitle.getPrefixedText(), captionTargetPageTitle.getWikiSite().languageCode(), captionTargetPageTitle,
-                                        captionTargetPageTitle.getDisplayText(), null, getLeadImageUrl());
-                                pageHeaderView.setUpCallToAction(app.getResources().getString(R.string.suggested_edits_article_cta_image_caption_in_language, app.language().getAppLanguageLocalizedName(lang)));
-                                break;
+                        ServiceFactory.get(new WikiSite(Service.COMMONS_URL)).getImageInfo(imageTitle, WikipediaApp.getInstance().getAppOrSystemLanguageCode()), Pair::new))
+                .flatMap(pair -> {
+                            captionSourcePageTitle = new PageTitle(imageTitle, new WikiSite(Service.COMMONS_URL, getTitle().getWikiSite().languageCode()));
+                            imageCaptions = pair.first;
+                            imageInfo = pair.second.query().firstPage().imageInfo();
+                            imagePage = pair.second.query().firstPage();
+                            if (!imageCaptions.containsKey(getTitle().getWikiSite().languageCode())) {
+                                imageEditType = ImageEditType.ADD_CAPTION;
+                                return ImageTagsProvider.getImageTagsObservable(pair.second.query().firstPage().pageId(), getTitle().getWikiSite().languageCode());
+                            }
+                            if (app.language().getAppLanguageCodes().size() >= MIN_LANGUAGES_TO_UNLOCK_TRANSLATION) {
+                                for (String lang : app.language().getAppLanguageCodes()) {
+                                    if (!imageCaptions.containsKey(lang)) {
+                                        imageEditType = ImageEditType.ADD_CAPTION_TRANSLATION;
+                                        captionTargetPageTitle = new PageTitle(imageTitle, new WikiSite(Service.COMMONS_URL, lang));
+                                        break;
+                                    }
+                                }
                             }
+                            return ImageTagsProvider.getImageTagsObservable(pair.second.query().firstPage().pageId(), getTitle().getWikiSite().languageCode());
                         }
+                )
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(imageTagsResult -> {
+                    if (imageEditType != ImageEditType.ADD_CAPTION && imageTagsResult != null && imageTagsResult.size() == 0) {
+                        imageEditType = ImageEditType.ADD_TAGS;
                     }
-                }, L::e));
+                    updateCta();
+                })
+        );
+    }
+
+    private void updateCta() {
```

Comment:
Can we have other name instead of `updateCta()` since the method above is `updateCallToAction()`? (e.g. `finalizeCallToAction()`